core/extensions/extension.hpp

```diff
@@ -20,87 +22,199 @@ namespace kagome::extensions {
         size_t state, size_t func_index)>;
 
    public:
-    /// storage extensions
-    virtual uint8_t *ext_child_storage_root(const uint8_t *storage_key_data,
-                                            uint32_t storage_key_length,
-                                            uint32_t *written) = 0;
+    // -------------------------Storage extensions--------------------------
 
-    virtual void ext_clear_child_storage(const uint8_t *storage_key_data,
-                                         uint32_t storage_key_length,
-                                         const uint8_t *key_data,
-                                         uint32_t key_length) = 0;
+    /**
+     * @brief Deletes values by keys containing given prefix
+     * @param prefix_data pointer to the prefix
+     * @param prefix_length lemgth of the prefix
+     */
+    virtual void ext_clear_prefix(runtime::WasmPointer prefix_data,
+                                  runtime::SizeType prefix_length) = 0;
 
-    virtual void ext_clear_prefix(const uint8_t *prefix_data,
-                                  uint32_t prefix_length) = 0;
+    /**
+     * @brief Deletes value by given key
+     * @param key_data pointer to the key
+     * @param key_length length of the key
+     */
+    virtual void ext_clear_storage(runtime::WasmPointer key_data,
+                                   runtime::SizeType key_length) = 0;
 
-    virtual void ext_clear_storage(const uint8_t *key_data,
-                                   uint32_t key_length) = 0;
+    /**
+     * @brief Checks if the given key exists in the storage.
+     * @param key_data pointer to the key
+     * @param key_length length of the key
+     * @return 1 if value exists, 0 otherwise
+     */
+    virtual runtime::SizeType ext_exists_storage(
+        runtime::WasmPointer key_data, runtime::SizeType key_length) = 0;
```

Comment:
By the way, can't some of the functions be made const? For example, this one?

Reply:
We should be really careful about that. Because some times result is written into wasm storage, which is not const operation. However, for that function and `chain_id` from misc extensions we can mar function as const